spec/webhook_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'spec_helper'
+
+describe Mailgun::Webhook do
+
+  before :each do
+    @mailgun = Mailgun({:api_key => "api-key", :webhook_url => "http://postbin.heroku.com/860bcd65"})
+
+    @sample = {
+      :id     => "click",
+      :url    => "http://postbin.heroku.com/860bcd65"
+    }
+  end
+
+  describe "list avabilable webhook ids" do
+    it "should return the correct ids" do
+      @mailgun.webhooks.available_ids.should =~ %w(bounce deliver drop spam unsubscribe click open).map(&:to_sym)
+    end
+  end
+
+  describe "list webhooks" do
+    it "should make a GET request with the correct params" do
+
+      sample_response = "{\"total_count\": 1, \"items\": [{\"webhooks\":{\"open\":{\"url\":\"http://postbin.heroku.com/860bcd65\"},\"click\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}}]}"
+      webhooks_url = @mailgun.webhooks.send(:webhook_url)
+
+      Mailgun.should_receive(:submit).
+        with(:get, webhooks_url).
+        and_return(sample_response)
+
+      @mailgun.webhooks.list
+    end
+  end
+
+  describe "find a webhook" do
+    it "should make a GET request with correct params to find a given webhook" do
+      sample_response = "{\"webhook\": {\"url\":\"http://postbin.heroku.com/860bcd65\"}"
+      webhooks_url = @mailgun.webhooks.send(:webhook_url, @sample[:id])
+
+      Mailgun.should_receive(:submit).
+        with(:get, webhooks_url).
+        and_return(sample_response)
+
+      @mailgun.webhooks.find(@sample[:id])
+    end
+  end
+
+  describe "add a webhook" do
+    context "using the default webhook url" do
+      it "should make a POST request with correct params to add a webhook" do
+        sample_response = "{\"message\":\"Webhook has been created\",\"webhook\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}"
+        webhooks_url = @mailgun.webhooks.send(:webhook_url)
+
+        Mailgun.should_receive(:submit).
+          with(:post, webhooks_url, {:id => @sample[:id], :url => @sample[:url]}).
```

Comment:
Line is too long. [82/80]<br>Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.